### PR TITLE
Adding bug dependencies as feature

### LINKED ISSUE #109: Add number of bug dependencies as a feature

### PR DESCRIPTION
**Pull request by parthpm**
Added Class for no of bug dependencies.
Fixes #109.

### CODE REVIEW

**Comment by ayush1999** [comment]
Run flake8 locally before pushing. (That's why the build is failing).

**Comment by ayush1999** [comment]
You still have flake8 errors. Did you try running it locally?

**Comment by marco-c** [comment]
Could you try to add this feature to the bug and the tracking models and see if it makes any difference (run the training both before/after adding the feature and post the results here).

**Comment by parthpm** [comment]
Sure,Can you state in which models/files I need to add this feature?

**Comment by marco-c** [comment]
> Sure,Can you state in which models/files I need to add this feature?

The bug and tracking models :smile:

**Comment by parthpm** [comment]
Before the feature:
Cross Validation scores:
Accuracy: f0.951746383784194 (+/- 0.0027559934918468167)
Precision: f0.9821515475460835 (+/- 0.0025632765014633823)
Recall: f0.9621308627678055 (+/- 0.002984157203642199)
After the feature:
Cross Validation scores:
Accuracy: f0.9537653622633995 (+/- 0.007353494408023867)
Precision: f0.9817444051612358 (+/- 0.005308488894346107)
Recall: f0.9649102489866822 (+/- 0.006933007317705039)

**Comment by marco-c** [comment]
Which model is this?

**Comment by parthpm** [comment]
default one

**Comment by marco-c** [comment]
So those are the results for the bug model. Please also post the results for the tracking model.

File: bugbug/models/bug.py [hunk 1]
```diff
@@ -41,6 +41,7 @@ def __init__(self, lemmatization=False):
             bug_features.affected_then_unaffected(),
             bug_features.product(),
             bug_features.component(),
+            bug_features.number_of_bug_dependencies(),
```
Comment (marco-c) [inline bugbug/models/bug.py]: Since there was no improvement for this model, we shouldn't add the feature here.

**Comment by parthpm** [comment]
For Tracking Model:
Before the feature:
Accuracy: f0.9267052832008851 (+/- 0.0076395288329126195)
Precision: f0.32845955697564244 (+/- 0.023050564418674844)
Recall: f0.9296703296703296 (+/- 0.00640379216777748)

After the feature:
Cross Validation scores:
Accuracy: f0.929059165703246 (+/- 0.0010861791237453567)
Precision: f0.33538382855727816 (+/- 0.002857861907866733)
Recall: f0.9284144427001569 (+/- 0.011337814810215247)

**Comment by marco-c** [comment]
Thanks! So let's keep it in the tracking model and remove it from the bug model.

**Pull request merged**